pkg/controlplane/control/manager.go

```diff
@@ -48,10 +70,62 @@ type Manager struct {
 func (m *Manager) AddImport(ctx context.Context, imp *v1alpha1.Import) error {
 	m.logger.Infof("Adding import '%s/%s'.", imp.Namespace, imp.Name)
 
-	newService := &v1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      imp.Name,
+	var err error
+	defer func() {
```

Comment:
q: why is the function deferred?

Reply:
because it manipulates the returned `err`.

Reply:
It could obscure the flow, typically `defer` is used for cleanup, whereas here it is used to manipulate the object.
Just to make the flow explicit, could the same be achieved using an `updateCondistions` func and calling it directly whenever `return err` is also called? It would also help in shortening this function somewhat from its current ~150 lines,

Reply:
using defer is the safest, since it always occurs, just like using a `finally` clause.
With your suggestion, a future programmer can fail to call `updateConditions` when a new return is introduced.
Also note that this is an accepted pattern even by the `nonamedreturnlinter` which we have enabled:
https://github.com/firefart/nonamedreturns
`Named errors used in defers are not reported.`